File: app/src/main/java/ru/job4j/todolist/ShowTaskActivity.java
package ru.job4j.todolist;

import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.squareup.picasso.Picasso;

import java.io.File;
import java.util.Objects;

import ru.job4j.todolist.store.FileStore;
import ru.job4j.todolist.store.IStore;

public class ShowTaskActivity extends AppCompatActivity {
    private IStore store;

    @Override
    protected void onCreate(@Nullable Bundle state) {
        super.onCreate(state);
        setContentView(R.layout.activity_show_task);
        Objects.requireNonNull(getSupportActionBar()).setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);
        int position = getIntent().getIntExtra("id for show", -1);
        this.store = FileStore.getInstance(this);
        Task task = store.findTaskByID(position);
        ((TextView) findViewById(R.id.show_task_name)).setText(task.getName());
        ((TextView) findViewById(R.id.show_task_description)).setText(task.getDesc());
        ((TextView) findViewById(R.id.show_task_created))
                .setText(task.getCreated());
        if (task.getClosed() != null){
            ((TextView) findViewById(R.id.show_task_finished))
                    .setVisibility(View.VISIBLE);
        }
        ((TextView) findViewById(R.id.show_task_closed))
                .setText(task.getClosed());
        ImageView photo = findViewById(R.id.show_task_photo);
        if(task.getPhoto() != null) {
            Picasso.with(this).load(new File(task.getPhoto())).into(photo);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            onBackPressed();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/ru/job4j/todolist/ShowTaskActivity.kt
package ru.job4j.todolist

import android.os.Bundle
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.activity_show_task.*
import ru.job4j.todolist.store.FileStore
import java.io.File

class ShowTaskActivity : AppCompatActivity() {

    override fun onCreate(state: Bundle?) {
        super.onCreate(state)
        setContentView(R.layout.activity_show_task)
        supportActionBar?.apply {
            setDisplayHomeAsUpEnabled(true)
            setDisplayShowHomeEnabled(true)
        }
        val position = intent.getIntExtra("id for show", -1)
        val (_, name, desc, created, closed, photo) =
                FileStore.getInstance(this).findTaskByID(position)!!
        show_task_name.text = name
        show_task_description.text = desc
        show_task_created.text = created
        show_task_finished.visibility = if(closed != null) View.VISIBLE else View.INVISIBLE
        show_task_closed.text = closed
        photo?.let {Picasso.with(this).load(File(photo)).into(show_task_photo)}
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            onBackPressed()
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}
